bin/partitions.toit

```diff
@@ -0,0 +1,255 @@
+    table.do: | partition/Partition |
+      type_string/string := ?
+      if partition.type == 0: type_string = "app"
+      else if partition.type == 1: type_string = "data"
+      else: type_string = "$partition.type"
+
+      k_size := partition.size / 1024
+      print "$partition.name, $type_string, $partition.subtype, 0x$(%x partition.offset), 0x$(%x partition.size), $(k_size)K"
+
+print_otadata parsed/cli.Parsed:
+  with_esptool parsed: | esptool/Esptool |
+    partition_table_bytes := esptool.read_partition_table
+    table := PartitionTable.decode partition_table_bytes
+    otadata_partition := table.find_otadata
+    otadata_offset := otadata_partition.offset
+    otadata_size := otadata_partition.size
+    otadata_bytes := esptool.read_flash
+        --offset=otadata_offset
+        --size=otadata_size
+
+    otadata := Otadata.decode otadata_bytes
+    2.repeat:
+      entry/SelectEntry := otadata.select_entries[it]
+      print """
+      otadata$it:
+        sequence-number: $entry.sequence_number
+        label: $entry.label.to_string_non_throwing
+        state: $entry.state ($(SelectEntry.state_stringify entry.state))
+        crc: $(%x entry.crc)"""
+      if it == 0: print
+
+read_partition parsed/cli.Parsed:
+  with_esptool parsed: | esptool/Esptool|
+    out := parsed["out"]
+    partition_name := parsed["partition"]
+
+    partition_table_bytes := esptool.read_partition_table
+    table := PartitionTable.decode partition_table_bytes
+    partition := table.find --name=partition_name
+    if not partition:
+      print "Partition '$partition_name' not found"
+      exit 1
+
+    esptool.read_flash
+        --offset=partition.offset
+        --size=partition.size
+        --out=out
+
+write_partition parsed/cli.Parsed:
+  with_esptool parsed: | esptool/Esptool |
+    in := parsed["in"]
+    partition_name := parsed["partition"]
+
+    partition_table_bytes := esptool.read_partition_table
+    table := PartitionTable.decode partition_table_bytes
+    partition := table.find --name=partition_name
+    if not partition:
+      print "Partition '$partition_name' not found"
+      exit 1
+
+    esptool.write_flash
+        --offset=partition.offset
+        --path=in
+
+set_ota_state parsed/cli.Parsed:
+  with_esptool parsed: | esptool/Esptool |
+    partition_name := parsed["partition"]
+    make_active := parsed["make-active"]
+    select_entry_index := parsed["select-entry"]
+
+    if partition_name != "ota_0" and partition_name != "ota_1":
+      print "Invalid partition name '$partition_name'"
+      exit 1
+
+    partition_table_bytes := esptool.read_partition_table
+    table := PartitionTable.decode partition_table_bytes
+    otadata_bytes := esptool.read_flash
+        --offset=table.find_otadata.offset
+        --size=table.find_otadata.size
+
+    otadata := Otadata.decode otadata_bytes
+
+    // For simplicity we always update entry 0 for ota_0 and entry 1 for ota_1.
+    // In theory
```

Comment:
Missing 2nd half of comment?

Reply:
removed the comment. Not even sure it made sense.
done.